commit 37004c42f7240035bc2726c340c4efa726b4818e
# use close_bdev_exclusive has been replaced by blkdev_put
Author: Tejun Heo <tj@kernel.org>
Date:   Sat Nov 13 11:55:17 2010 +0100

    btrfs: close_bdev_exclusive() should use the same @flags as the matching open_bdev_exclusive()
    
    In the failure path of __btrfs_open_devices(), close_bdev_exclusive()
    is called with @flags which doesn't match the one used during
    open_bdev_exclusive().  Fix it.
    
    Signed-off-by: Tejun Heo <tj@kernel.org>
    Cc: Chris Mason <chris.mason@oracle.com>

diff --git a/fs/btrfs/volumes.c b/fs/btrfs/volumes.c
index cc04dc1..d395962 100644
--- a/fs/btrfs/volumes.c
+++ b/fs/btrfs/volumes.c
@@ -638,7 +638,7 @@ static int __btrfs_open_devices(struct btrfs_fs_devices *fs_devices,
 error_brelse:
 		brelse(bh);
 error_close:
-		close_bdev_exclusive(bdev, FMODE_READ);
+		close_bdev_exclusive(bdev, flags);
 error:
 		continue;
 	}


